opal/corelib/number.rb

```diff
@@ -261,6 +261,19 @@ def **(other)
     end
   end
 
+  def ===(other)
+    %x{
+      if (other.$$is_number) {
+        if (typeof other === 'number') {
+          return self.valueOf() === other;
+        }
+        return self.valueOf() === Number(other);
```

Comment:
Could you benchmark this without the `typeof` check and just `self.valueOf() === other.valueOf()`?